src/integer_mod_q/mat_zq/from.rs

```diff
@@ -130,9 +100,31 @@ impl FromStr for MatZq {
 }
 
 impl<Mod: Into<Modulus>> From<(&MatZ, Mod)> for MatZq {
-    /// Convert [`MatZ`] and [`Modulus`] to [`MatZq`] using [`MatZq::from_mat_z_modulus`].
+    /// Create a [`MatZq`] from a [`MatZ`] and a [`Modulus`].
+    ///
+    /// Parameters:
+    /// - `matrix`: the matrix from which the entries are taken
+    /// - `modulus`: the modulus of the matrix
+    ///
+    /// Returns the new matrix.
+    ///
+    /// # Examples
+    /// ```
+    /// use qfall_math::integer::MatZ;
+    /// use qfall_math::integer_mod_q::MatZq;
+    /// use std::str::FromStr;
+    ///
+    /// let m = MatZ::from_str("[[1, 2],[3, -1]]").unwrap();
+    ///
+    /// let a = MatZq::from((&m, 17));
+    /// ```
     fn from((matrix, modulus): (&MatZ, Mod)) -> Self {
-        MatZq::from_mat_z_modulus(matrix, modulus)
+        let mut out = MatZq::new(matrix.get_num_rows(), matrix.get_num_columns(), modulus);
+        unsafe {
+            fmpz_mat_set(&mut out.matrix.mat[0], &matrix.matrix);
+            _fmpz_mod_mat_reduce(&mut out.matrix);
```

Comment:
Consider changing it to `fmpz_mod_mat_set_fmpz_mat` (should do the same and we have less interface usage)